api/Port.go

```diff
@@ -151,3 +177,98 @@ func (vrsConnection *VRSConnection) UpdatePortMetadata(name string, metadata map
 
 	return nil
 }
+
+// GetNuagePortTableUpdate will register with OVSDB
+// for Nuage Port table updates and return as soon as
+// port table entry gets populated
+func (vrsConnection *VRSConnection) RegisterForPortUpdates(brport string, pnc chan *PortIPv4Info) error {
+	vrsConnection.registrationChannel <- &Registration{Brport: brport, Channel: pnc, Register: true}
+	return nil
+}
+
+func (vrsConnection *VRSConnection) DeregisterForPortUpdates(brport string) error {
+	vrsConnection.registrationChannel <- &Registration{Brport: brport, Channel: nil, Register: false}
+	return nil
+}
+
+func (vrsConnection VRSConnection) handlePortRegistration(registration *Registration) error {
+	brport := registration.Brport
+	register := registration.Register
+	pnc := registration.Channel
+	if register {
+		if _, ok := vrsConnection.pncTable[brport]; ok {
+			return fmt.Errorf("Already registered for this bridge port", brport)
+		} else {
+			vrsConnection.pncTable[brport] = pnc
+			if portInfo, exists := vrsConnection.pnpTable[brport]; exists {
+				pnc <- &portInfo
+				delete(vrsConnection.pnpTable, brport)
+			}
+		}
+	} else {
+		delete(vrsConnection.pncTable, brport)
+	}
+	return nil
+}
+
+func (vrsConnection VRSConnection) getPortInfo(row *libovsdb.Row) (*PortIPv4Info, error) {
+	portIPv4Info := PortIPv4Info{Registered: true}
+	if _, ok := row.Fields["ip_addr"]; ok {
+		ip := row.Fields["ip_addr"].(string)
+		if ip != "" {
+			portIPv4Info.IPAddr = ip
+		} else {
+			return nil, errors.New("Invalid or empty ip")
+		}
+	}
+	if _, ok := row.Fields["subnet_mask"]; ok {
+		subnet := row.Fields["subnet_mask"].(string)
+		if subnet != "" {
+			portIPv4Info.Mask = subnet
+		} else {
+			return nil, errors.New("Invalid or empty subnet")
+		}
+	}
+	if _, ok := row.Fields["gateway"]; ok {
+		gw := row.Fields["gateway"].(string)
+		if gw != "" {
+			portIPv4Info.Gateway = gw
+		} else {
+			return nil, errors.New("Invalid or empty gateway")
+		}
+	}
+	return &portIPv4Info, nil
+}
+
+func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates) error {
+	for _, tableUpdate := range updates.Updates {
+		for _, row := range tableUpdate.Rows {
+			empty := libovsdb.Row{}
+			if !reflect.DeepEqual(row.New, empty) {
+				//check for whether the port is already registered for updates
+				portInfo, err := vrsConnection.getPortInfo(&(row.New))
+				if err == nil {
+					if _, ok := (row.New).Fields["name"]; ok {
+						portName := (row.New).Fields["name"].(string)
+						if pncChannel, exists := vrsConnection.pncTable[portName]; exists {
+							pncChannel <- portInfo
```

Comment:
Should this be run in a go routine. What is the guy who registered for updates failed to listen on the channel? In that case this function will hang.